File: frontend/src/types/analysis.ts
export interface QuestionData {
  question_id: string
  question_text: string
  question_type: 'multiple_choice' | 'true_false' | 'short_answer' | 'essay' | 'unknown'
  topic: string
  difficulty?: 'easy' | 'medium' | 'hard'
  confidence_score: number
}

export interface TopicData {
  topic_id?: string
  topic_name: string
  question_count: number
  keywords?: string[]
  confidence_score?: number
}

export interface ClusterData {
  cluster_id: string
  questions: string[]
  size: number
}

export interface AnalysisData {
  questions: QuestionData[]
  topics: TopicData[]
  clusters: ClusterData[]
  summary: {
    total_questions: number
    topics_found: number
  }
}

export interface AnalysisResult {
  document_id: string
  filename?: string
  status: string
  questions_found: number
  topics_identified: number
  analysis_data: AnalysisData
}

File: backend/app/schemas/analysis.py
from pydantic import BaseModel
from typing import Dict, Any, List, Optional


class AnalysisRequest(BaseModel):
    document_id: str
    analysis_type: Optional[str] = "full"


class QuestionData(BaseModel):
    question_id: str
    question_text: str
    question_type: str
    topic: str
    difficulty: Optional[str]
    confidence_score: float


class AnalysisResponse(BaseModel):
    document_id: str
    status: str
    questions_found: int
    topics_identified: int
    analysis_data: Dict[str, Any]


class TopicCluster(BaseModel):
    topic_id: str
    topic_name: str
    question_count: int
    keywords: List[str]
    confidence_score: float